modules/nw-kuryr-cleanup.adoc

```diff
@@ -0,0 +1,280 @@
+        subports+=("$subport");
+    done
+    args=()
+    for sub in "${subports[@]}" ; do
+        args+=("--subport $sub")
+    done
+    if [ ${#args[@]} -gt 0 ]; then
+        openstack network trunk unset ${args[*]} $trunk
+    fi
+done
+----
+
+. To retrieve all networks and subnets from `KuryrNetwork` CRs and remove ports, router interfaces and the network itself, enter the following command:
++
+[source,terminal]
+----
+$ mapfile -t kuryrnetworks < <(oc get kuryrnetwork -A --template='{{range $i,$p := .items}}{{ $p.status.netId }}|{{ $p.status.subnetId }}{{"\n"}}{{end}}')
+$ i=0
+$ for kn in "${kuryrnetworks[@]}"; do
+    i=$((i+1))
+    netID=${kn%%|*}
+    subnetID=${kn##*|}
+    echo "Processing network $netID, ${i}/${#kuryrnetworks[@]}"
+    # Remove all ports from the network.
+    for port in $(python -c "import openstack; n = openstack.connect().network; print(' '.join([x.id for x in n.ports(network_id='$netID') if x.device_owner != 'network:router_interface']))"); do
+        ( openstack port delete $port ) &
+
+        # Only allow 20 jobs in parallel.
+        if [[ $(jobs -r -p | wc -l) -ge 20 ]]; then
+            wait -n
+        fi
+    done
+    wait
+
+    # Remove the subnet from the router.
+    openstack router remove subnet $ROUTERID $subnetID
+
+    # Remove the network.
+    openstack network delete $netID
+done
+----
+
+. To remove the Kuryr security group, enter the following command:
++
+[source,terminal]
+----
+$ openstack security group delete ${CLUSTERID}-kuryr-pods-security-group
+----
+
+. To remove all tagged subnet pools, enter the following command:
++
+[source,terminal]
+----
+$ for subnetpool in $(openstack subnet pool list --tags $CLUSTERTAG -f value -c ID); do
+    openstack subnet pool delete $subnetpool
+done
+----
+
+. To check that all of the networks based on `KuryrNetwork` CRs were removed, enter the following command:
++
+[source,terminal]
+----
+$ networks=$(oc get kuryrnetwork -A --no-headers -o custom-columns=":status.netId")
+for existingNet in $(openstack network list --tags $CLUSTERTAG -f value -c ID); do
+    if [[ $networks =~ $existingNet ]]; then
+        echo "Network still exists: $existingNet"
+    fi
+done
+----
++
+If the command returns any existing networks, intestigate and remove them before you continue.
+
+. To remove security groups that are related to network policy, enter the following command:
++
+[source,terminal]
+----
+$ for sgid in $(openstack security group list -f value -c ID -c Description | grep 'Kuryr-Kubernetes Network Policy' | cut -f 1 -d ' '); do
+    openstack security group delete $sgid
+done
+----
+
+. To remove finalizers from `KuryrNetwork` CRs, enter the following command:
++
+[source,terminal]
+----
+$ REMFIN kuryrnetworks.openstack.org kuryrnetwork.finalizers.kuryr.openstack.org
+----
+
+. If the your router *was not* created by the installer, enter the following command to remove the router:
```

Comment:
I've reshaped this a bit to indicate that in case router was created by installer, it shouldn't be touched

Reply:
```suggestion
. If the installer did not create your router, enter the following command to remove the router:
+
IMPORTANT: If the installer did create your router, do not remove the router.
```